src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -151,6 +154,75 @@
     </ItemGroup>
   </Target>
 
+  <!-- #################################### -->
+  <!-- Generate BuildVersion.props -->
+  <!-- #################################### -->
+  <Target Name="CreateOrUpdateCurrentVersionFile" Condition="'$(SkipVersionGeneration)'!='true'" DependsOnTargets="GenerateCurrentVersion">
+    <Error Condition="'$(OverrideBuildNumberMajor)'==''" Text="Error when generating current version." />
+
+    <ItemGroup>
+      <CurrentVersionLines Include="%3C%3Fxml version=%221.0%22 encoding=%22utf-8%22%3F%3E" />
+      <CurrentVersionLines Include="%3C!-- This is a generated file. To re-generate it, run build.cmd --%3E" />
+      <CurrentVersionLines Include="%3CProject xmlns=%22http://schemas.microsoft.com/developer/msbuild/2003%22%3E" />
+      <CurrentVersionLines Include="%3CPropertyGroup%3E" />
+      <CurrentVersionLines Include="%3CBuildNumberMajor Condition=%22%27%24(BuildNumberMajor)%27==%27%27%22%3E$(OverrideBuildNumberMajor)%3C/BuildNumberMajor%3E" />
+      <CurrentVersionLines Include="%3C/PropertyGroup%3E" />
+      <CurrentVersionLines Include="%3C/Project%3E" />
+      <BuildVersionFileItem Include="$(BuildVersionFile)" />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <BuildVersionSemaphoreFile>%(BuildVersionFileItem.RootDir)%(BuildVersionFileItem.Directory)semaphore.txt</BuildVersionSemaphoreFile>
```

Comment:
Actually if you do switch to inputs/outputs then you will need a semaphore file but I'm not sure there is much value in doing having it here. I think I would just do the exists check on the target condition itself for the buildversion.props file and if it doesn't exist execute the target. Just make sure the WriteLinesToFile correctly handles errors. I will tell you WarnAndContinue will cause build warnings from time to time and we will need to explain. I wonder if just doing nothing is better and simply check for existence of the file after the write and if it doesn't exist give an error.